lib/src/yaml_edit/editor.dart

```diff
@@ -0,0 +1,651 @@
+    final keyOrIndex = pathAsList.last;
+    final parentNode = _traverse(collectionPath, checkAlias: true);
+
+    final valueNode = wrapAsYamlNode(value);
+
+    if (parentNode is YamlList) {
+      final expectedList =
+          updatedYamlList(parentNode, (nodes) => nodes[keyOrIndex] = valueNode);
+
+      _performEdit(assignInList(this, parentNode, keyOrIndex, value),
+          collectionPath, expectedList);
+      return;
+    }
+
+    if (parentNode is YamlMap) {
+      final keyNode = wrapAsYamlNode(keyOrIndex);
+
+      final expectedMap =
+          updatedYamlMap(parentNode, (nodes) => nodes[keyNode] = valueNode);
+      _performEdit(assignInMap(this, parentNode, keyOrIndex, value),
+          collectionPath, expectedMap);
+      return;
+    }
+
+    throw PathError.unexpected(
+        path, 'Scalar $parentNode does not have key $keyOrIndex');
+  }
+
+  /// Appends [value] into the list at [path].
+  ///
+  /// Throws if the element at the given path is not a [YamlList] or if the path
+  /// is invalid.
+  void appendToList(Iterable<Object> path, Object value) {
+    ArgumentError.checkNotNull(path, 'path');
+    final yamlList = _traverseToList(path);
+
+    insertIntoList(path, yamlList.length, value);
+  }
+
+  /// Prepends [value] into the list at [path].
+  ///
+  /// Throws if the element at the given path is not a [YamlList] or if the path
+  /// is invalid.
+  void prependToList(Iterable<Object> path, Object value) {
+    ArgumentError.checkNotNull(path, 'path');
+
+    insertIntoList(path, 0, value);
+  }
+
+  /// Inserts [value] into the list at [path], only if the element at the given
+  /// path is a list.
+  ///
+  /// [index] must be non-negative and no greater than the list's length.
+  ///
+  /// Throws if the element at the given path is not a [YamlList] or if the path
+  /// is invalid.
+  void insertIntoList(Iterable<Object> path, int index, Object value) {
+    ArgumentError.checkNotNull(path, 'path');
+
+    final list = _traverseToList(path, checkAlias: true);
+    RangeError.checkValueInInterval(index, 0, list.length);
+
+    final edit = insertInList(this, list, index, value);
+
+    final expectedList = updatedYamlList(
+        list, (nodes) => nodes.insert(index, wrapAsYamlNode(value)));
+
+    _performEdit(edit, path, expectedList);
+  }
+
+  /// Changes the contents of the list at [path] by removing [deleteCount] items
+  /// at [index], and inserts [values] in-place.
+  ///
+  /// [index] and [deleteCount] must be non-negative and, [index] + [deleteCount]
+  /// must be no greater than the list's length. Throws otherwise or if [path]
+  /// is invalid.
+  ///
+  /// ```dart
+  /// final doc = YamlEditor('[Jan, March, April, June]');
+  /// doc.spliceList([], 1, 0, ['Feb']); // [Jan, Feb, March, April, June]
+  /// doc.spliceList([], 4, 1, ['May']); // [Jan, Feb, March, April, May]
+  /// ```
+  Iterable<YamlNode> spliceList(Iterable<Object> path, int index,
```

Comment:
Should this return `void` ?
Otherwise describe the returned value

Reply:
I chose to add a return value to make it similar to the Javascript [splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice), and also thought that doing so would make it behave similar to how a remove method works on maps.

I've added description for the return value for now, but if you think we should make it return void I can do that too!